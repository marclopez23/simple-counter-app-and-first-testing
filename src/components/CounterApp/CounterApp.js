import React, { useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handeleAdd = () => {
    setCounter((state) => (state += 1));
  };

  const handeleSubstract = () => {
    setCounter((state) => (state -= 1));
  };

  const handeleReset = () => {
    setCounter(value);
  };

  return (
    <section>
      <h1>Counter App</h1>
      <h2>{counter}</h2>
      <button onClick={() => handeleAdd()}> +1 </button>
      <button onClick={() => handeleSubstract()}> -1 </button>
      <button onClick={() => handeleReset()}> Reset </button>
    </section>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

CounterApp.defaultProps = {
  value: 0,
};

export default CounterApp;
