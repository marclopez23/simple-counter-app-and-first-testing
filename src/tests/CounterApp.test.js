import React from "react";
import { shallow } from "enzyme";
import CounterApp from "../components/CounterApp/CounterApp";

describe("Pruebas en CounterApp", () => {
  let wrapper = shallow(<CounterApp />);
  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  });
  test("CounterApp se muestra correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("CounterApp recibe bien las props", () => {
    const value = 100;
    const wrapper = shallow(<CounterApp value={value} />);
    const valuePrinted = wrapper.find("h2").text().trim();

    expect(valuePrinted).toBe(value.toString());
  });

  test("debe de incrementar con el boton +1", () => {
    wrapper.find("button").at(0).simulate("click");
    const valuePrinted = wrapper.find("h2").text().trim();

    expect(valuePrinted).toBe("1");
  });

  test("debe de restar con el boton -1", () => {
    wrapper.find("button").at(1).simulate("click");
    const valuePrinted = wrapper.find("h2").text().trim();
    expect(valuePrinted).toBe("-1");
  });

  test("debe de resetear el valor", () => {
    const value = 105;
    const wrapper = shallow(<CounterApp value={value} />);
    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(2).simulate("click");
    const valuePrinted = wrapper.find("h2").text().trim();
    expect(valuePrinted).toBe(value.toString());
  });
});
