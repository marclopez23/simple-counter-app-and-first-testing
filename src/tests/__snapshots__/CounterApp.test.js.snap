// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CounterApp CounterApp se muestra correctamente 1`] = `
<section>
  <h1>
    Counter App
  </h1>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
     +1 
  </button>
  <button
    onClick={[Function]}
  >
     -1 
  </button>
  <button
    onClick={[Function]}
  >
     Reset 
  </button>
</section>
`;
